DataFormats/MuonReco/src/MuonSelectors.cc

```diff
@@ -929,3 +949,48 @@ int muon::sharedSegments( const reco::Muon& mu, const reco::Muon& mu2, unsigned
   
     return ret; 
 }
+
+void muon::setCutBasedSelectorFlags(reco::Muon& muon, 
+				    const reco::Vertex* vertex,
+				    bool run2016_hip_mitigation)
+{
+  // https://twiki.cern.ch/twiki/bin/viewauth/CMS/SWGuideMuonIdRun2
+  unsigned int selectors = muon.selectors();
+  // Compute Id and Isolation variables
+  double chIso  = muon.pfIsolationR04().sumChargedHadronPt;
+  double nIso   = muon.pfIsolationR04().sumNeutralHadronEt;
+  double phoIso = muon.pfIsolationR04().sumPhotonEt;
+  double puIso  = muon.pfIsolationR04().sumPUPt;
+  double dbCorrectedIsolation = chIso + std::max( nIso + phoIso - .5*puIso, 0. ) ;
+  double dbCorectedRelIso = dbCorrectedIsolation/muon.pt();
+  double tkRelIso = muon.isolationR03().sumPt/muon.pt();
+
+  // Base selectors
+  if (muon::isLooseMuon(muon))        selectors |= reco::Muon::CutBasedIdLoose;
+  if (vertex){
+    if (muon::isTightMuon(muon,*vertex))  selectors |= reco::Muon::CutBasedIdTight;
+    if (muon::isSoftMuon(muon,*vertex,run2016_hip_mitigation))   selectors |= reco::Muon::SoftCutBasedId;
+    if (muon::isHighPtMuon(muon,*vertex)) selectors |= reco::Muon::CutBasedIdGlobalHighPt;
+    if (muon::isTrackerHighPtMuon(muon,*vertex)) selectors |= reco::Muon::CutBasedIdTrkHighPt;
+  }
+  if (muon::isMediumMuon(muon,run2016_hip_mitigation)){
+    selectors |= reco::Muon::CutBasedIdMedium;
+    if ( vertex and 
+	 fabs(muon.muonBestTrack()->dz( vertex->position()))<0.1 and 
+	 fabs(muon.muonBestTrack()->dxy(vertex->position()))< 0.02 )
+      selectors |= reco::Muon::CutBasedIdMediumPrompt;
+  }
+
+  // PF isolation
+  if (dbCorectedRelIso<0.40)    selectors |= reco::Muon::PFIsoVeryLoose;
```

Comment:
another place where each check selects only a subset of the previous check. Please take advantage of that.

Reply:
That's one of the few examples where one selector is strictly a subset of the other by construction.